detectors/resources/src/main/java/com/google/cloud/opentelemetry/detectors/GCPMetadataConfig.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.detectors;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Retrieves Google Cloud project-id and a limited set of instance attributes from Metadata server.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/storing-retrieving-metadata">
+ *     https://cloud.google.com/compute/docs/storing-retrieving-metadata</a>
+ */
+final class GCPMetadataConfig {
```

Comment:
Is this class meant to be called many times in the lifecycle of a Java program?

Right now every call to any method will repeatedly hit the HTTP metadata server endpoint.  It'd be nicer if, instead, you had a method that collected all the components of information, issued ONE failure message if it had issues, and then filled out this MetadataConfig class.


Right now GCPMetadataConfig looks like a static, immutable instance.   Ideally make `GCPMetadataConfig` immutable with one static "fetch" *or* you'd rename all these "getXYZ" to "fetchXYZ" and rename the class to `GCPMetadataService`.

Reply:
What's the desired behavior if the metadata endpoint times out? Is there ever a scenario where we want to retry the fetch?

Reply:
For example, do we want to refresh instance/zone info after live migration? (This should be covered in our ops spec, not necessarily a change to this PR).